browser/src/core/Socket.js

```diff
@@ -1198,7 +1197,7 @@ app.definitions.Socket = L.Class.extend({
 			this._onHyperlinkClickedMsg(textMsg);
 		}
 
-		if (!this._isReady() || !this._map._docLayer || this._delayedMessages.length || this._handlingDelayedMessages) {
+		if (!this._map._docLayer) {
```

Comment:
Are we sure not using _isReady is a good idea ? =) if we don't want it - then we should prolly remove the function - since this is it's whole raison d'etre I think =)

Reply:
The this._map._docLayer is set in _onStatusMsg which is also using _isReady() (admittedly with another of these "try again later" timers) so _isReady() has to be true if this._map._docLayer is set.

Reply:
I guess things like:

commit a93d8f1e1cadd6a80248f8e6cb9a8eae80f0b657
Author: Ashod Nakashian <ashod.nakashian@collabora.co.uk>
Date:   Wed Jul 3 00:29:43 2019 -0400

Suggest 'isReady' is useful in some cases. Bear in mind that we can start executing JS without loading all of it (I think) in the browser, and its order of initialization is not cleanly defined.

Then again the _isReady in _onStatusMsg seems a bit frightening - but the browser really can take quite some time to get its act together, and its not entirely predictable when that happens (I expect).

Not sure that overall this is a good idea. I forget what the goal was here - a  better understanding? removing a long timeout (100ms is a good one to loose of course =), or ? ...

Reply:
The original thing that invited scrutiny was  https://github.com/CollaboraOnline/online/issues/7164 and during that a question was, why only delay certain messages if they can't be applied yet, so we now delay all messages until they can be applied and not lose any.

But a follow up was to see if we really need another long timeout to apply them. Which is this proposed change which takes advantage that this._map._docLayer gets set when _isReady() is true to drop that timer.

Reply:
Ah! ok - so I missed that _onStatusMsg is also doing a similar timeout spin (and we should prolly shorten the timeout to 10ms or so there) to check that we are really ready - and when done so it sets the ... ok :-) I get it.

And we're confident that there is always a _onStatusMsg coming through - since this is IIRC one of the very earliest messages coming in ...

Good :-) nice cleanup - sorry to miss that bit!.